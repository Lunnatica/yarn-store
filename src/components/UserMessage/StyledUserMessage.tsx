import styled from 'styled-components';

import { getTextStyles } from '../../theme/typography';
import { visuallyHidden } from '../StyledVisuallyHiddenComponents';

export const StyledUserMessage = styled.p<{
  visible: boolean;
  type: 'error' | 'info';
}>`
  /* Visually hide the component while keeping it in the DOM */
  ${visuallyHidden}

  /* Show the component if it has content */
  ${({ visible }) => visible && 'position: static;'}

  ${getTextStyles(600, '24px', '24px')}
  padding: 24px;
  border-radius: 8px;
  text-align: center;
  margin: 80px auto;
  width: 75%;
  color: ${({ type }) => (type === 'error' ? 'var(--colors-blank)' : 'var(--colors-darkBlue)')};
  background-color: ${({ type }) =>
    type === 'error' ? 'var(--colors-error)' : 'var(--colors-ghostWhite)'};
`;
