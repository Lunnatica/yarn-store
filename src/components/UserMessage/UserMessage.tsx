import { StyledUserMessage } from './StyledUserMessage';

interface UserMessageProps {
  className?: string;
  message?: string;
  type?: 'error' | 'info';
}

export const UserMessage: React.FC<UserMessageProps> = ({
  className,
  message = '',
  type = 'info',
}) => (
  <StyledUserMessage
    className={className}
    visible={message !== ''}
    type={type}
    aria-live={type === 'error' ? 'assertive' : 'polite'}
  >
    {message}
  </StyledUserMessage>
);
