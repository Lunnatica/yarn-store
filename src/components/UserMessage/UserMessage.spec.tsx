import { render, screen } from '@testing-library/react';
import { UserMessage } from './UserMessage';

describe('UserMessage', () => {
  it('should render the message', () => {
    render(<UserMessage message="Test message" />);
    const message = screen.getByText('Test message');
    expect(message);
  });

  it('should match the snapshot for error message', () => {
    const { container } = render(<UserMessage message="Error message" type="error" />);
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot for info message', () => {
    const { container } = render(<UserMessage message="Info message" type="info" />);
    expect(container).toMatchSnapshot();
  });
});
