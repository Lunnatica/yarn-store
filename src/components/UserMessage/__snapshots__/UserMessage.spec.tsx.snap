// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMessage should match the snapshot for error message 1`] = `
<div>
  <p
    aria-live="assertive"
    class="StyledUserMessage-sc-16zzyai-0 eHVwuF"
    type="error"
  >
    Error message
  </p>
</div>
`;

exports[`UserMessage should match the snapshot for info message 1`] = `
<div>
  <p
    aria-live="polite"
    class="StyledUserMessage-sc-16zzyai-0 iAnMCV"
    type="info"
  >
    Info message
  </p>
</div>
`;
